### ISSUE TITLE
memorycard not in the right format?

### ISSUE THREAD

**Issue by CodeAsm** [arkwise/blackroolinux#3]
initrd/blackroo_initrd.zip
these dont seem to be formatted correctly for what the kernel expects.

Yours begin with 0xEB3C, while the kernel expects to see a magic header of 0x3412 (bu id)

I was trying to write a memory card prepare tool and altho im not exactly sure these are right, based on the kernel source i made this struct:
```c
    // Prepare the header
    bu_first_block_t header = {
        0x1234,       // id
        131072,       // size
        0x00010000,   // serial
        3//0x01000000    // number
    };
```
you can find a preliminary test program and source here: https://github.com/CodeAsm/PS1Linux/blob/main/tools/psx-mcard/makemu.c
It will sadly not yet explain where the partition needs to start (I guess and hope the next sector?) for mutliple cards, you definitly need to play arround with the size, and number. (number determines the order you insert them in the ps1) the id stays the same, and its different from regular memorycards.

**Comment by arkwise** [comment]
Yes you are correct this is just a preliminary idea for it, im actually looking at ps2mcfs for answers to add a layer to expand its capability to read the psx memory cards on the ps2 linux i am running. i was trying to open some .mcr files that are exactly 131072 bytes in size to later use my Pi-mem setup later on also.

some interesting sources i found around:
 http://www.debaser.force9.co.uk/psxpblib/ has a library that can offer help via the parallel port and low level spi.
 
https://github.com/raphnet/psxmemcardmgr/ , https://github.com/brad-lin/FreePSXBoot so that we can create the correct images and interfaces to get this working. 
 
To make ps2mcfs work with PSX memory cards on Linux for the PS2. PSXPBlib shows how to talk to the memory cards at a low level, like reading and writing data directly. PSX Memory Card Manager helps us understand how the files and saves are organized on the card, so we can make that data usable.
 
 By using what we learn from these projects, we can add support for PSX memory cards to ps2mcfs, making it easy to read and write them on a PS2 running Linux.
 
 i have the multi taps ready for a raid setup, several ps2 and psx consoles
 
 its only basic research as i have been busy with other projects but ill get back to it eventually.
